src/dotgraph.cpp

```diff
@@ -0,0 +1,354 @@
+*  The .md5 is created or updated after successful creation of the output file.
+*/
+static bool checkMd5Signature(const QCString &baseName,
+  const QCString &md5)
+{
+  QFile f(baseName+".md5");
+  if (f.open(IO_ReadOnly))
+  {
+    // read checksum
+    QCString md5stored(33);
+    int bytesRead=f.readBlock(md5stored.rawData(),32);
+    md5stored[32]='\0';
+    // compare checksum
+    if (bytesRead==32 && md5==md5stored)
+    {
+      // bail out if equal
+      return FALSE;
+    }
+  }
+  f.close();
+  return TRUE;
+}
+
+static bool checkDeliverables(const QCString &file1,
+  const QCString &file2=QCString())
+{
+  bool file1Ok = TRUE;
+  bool file2Ok = TRUE;
+  if (!file1.isEmpty())
+  {
+    QFileInfo fi(file1);
+    file1Ok = (fi.exists() && fi.size()>0);
+  }
+  if (!file2.isEmpty())
+  {
+    QFileInfo fi(file2);
+    file2Ok = (fi.exists() && fi.size()>0);
+  }
+  return file1Ok && file2Ok;
+}
+
+static void removeDotGraph(const QCString &dotName)
+{
+  if (DOT_CLEANUP)
+  {
+    QDir d;
+    d.remove(dotName);
+  }
+}
+
+static bool insertMapFile(FTextStream &out,const QCString &mapFile,
+  const QCString &relPath,const QCString &mapLabel)
+{
+  QFileInfo fi(mapFile);
+  if (fi.exists() && fi.size()>0) // reuse existing map file
+  {
+    QGString tmpstr;
+    FTextStream tmpout(&tmpstr);
+    convertMapFile(tmpout,mapFile,relPath,FALSE);
+    if (!tmpstr.isEmpty())
+    {
+      out << "<map name=\"" << mapLabel << "\" id=\"" << mapLabel << "\">" << endl;
+      out << tmpstr;
+      out << "</map>" << endl;
+    }
+    return TRUE;
+  }
+  return FALSE; // no map file yet, need to generate it
+}
+
+//--------------------------------------------------------------------
+
+QCString DotGraph::IMG_EXT;
+
+QCString DotGraph::imgName() const 
+{
+  return m_baseName + ((m_graphFormat == GOF_BITMAP) ? ("." + IMG_EXT) : (USE_PDFLATEX ? ".pdf" : ".eps"));
+}
+
+QCString DotGraph::writeGraph(
+  FTextStream& t,           // output stream for the code file (html, ...)
+  GraphOutputFormat gf,     // bitmap(png/svg) or ps(eps/pdf)
+  EmbeddedOutputFormat ef,  // html, latex, ...
+  const char* path,         // output folder
+  const char* fileName,     // name of the code file (for code patcher)
+  const char* relPath,      // output folder relativ to code file
+  bool generateImageMap,    // in case of bitmap, shall there be code generated?
+  int graphId               // number of this graph in the current code, used in svg code
+)
+{
+  m_out = &t;
+  m_graphFormat = gf;
+  m_textFormat = ef;
+  m_dir = QDir(path);
+  m_fileName = fileName;
+  m_relPath = relPath;
+  m_generateImageMap = generateImageMap;
+  m_graphId = graphId;
+
+  m_absPath  = QCString(m_dir.absPath().data()) + "/";
+  m_baseName = getBaseName();
+
+  computeTheGraph();
+
+  m_regenerate = prepareDotFile();
+
+  if (!m_doNotAddImageToIndex) Doxygen::indexList->addImageFile(imgName());
+
+  generateCode();
+  //todo

```

Comment:
I'm looking at integrating your pull request and I was wondering what needs to be done here?

Reply:
the //todo needs to be removed ;)